### PR TITLE
[2.5] Bump k8s libraries for skippy bug

### PR DESCRIPTION
**Pull request by thedadams**
There is a CVE in gogo/protobuf that affects the kubernetes libraries.
Those libraries are upgraded to v0.21.0, which is not affected by the
CVE.

### CODE REVIEW

**Review by nickgerace — DISMISSED** [review]

**Review by prachidamle — APPROVED** [review]

**Review by nickgerace — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -13,31 +13,31 @@ replace (
...
-	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.20.0
-	k8s.io/apimachinery => k8s.io/apimachinery v0.20.0
-	k8s.io/apiserver => k8s.io/apiserver v0.20.0
-	k8s.io/cli-runtime => k8s.io/cli-runtime v0.20.0
+	k8s.io/api => k8s.io/api v0.20.6
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.20.6
+	k8s.io/apimachinery => k8s.io/apimachinery v0.20.6
+	k8s.io/apiserver => k8s.io/apiserver v0.20.6
+	k8s.io/cli-runtime => k8s.io/cli-runtime v0.20.6
 	k8s.io/client-go => github.com/rancher/client-go v1.20.0-rancher.1
```
Comment (StrongMonkey) [inline go.mod:21]: Do we need to bump client-go?

File: go.mod [hunk 2]
```diff
@@ -13,31 +13,31 @@ replace (
 	google.golang.org/grpc => google.golang.org/grpc v1.29.1 // etcd depends on google.golang.org/grpc/naming which was removed in grpc v1.30.0
 	helm.sh/helm/v3 => github.com/rancher/helm/v3 v3.3.0-rancher1
 
-	k8s.io/api => k8s.io/api v0.20.0
-	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.20.0
-	k8s.io/apimachinery => k8s.io/apimachinery v0.20.0
-	k8s.io/apiserver => k8s.io/apiserver v0.20.0
-	k8s.io/cli-runtime => k8s.io/cli-runtime v0.20.0
+	k8s.io/api => k8s.io/api v0.20.6
```
Comment (cbron) [inline go.mod:16]: Did you look to see if there are any breaking api changes between 0.20.0 and 0.20.6 ?

Comment (thedadams) [reply go.mod:16]: There are no listed breaking changes between these two versions.

File: go.mod (hunk 1 above)
Comment (thedadams) [reply go.mod:21]: As far as I remember, Kubernetes maintains patch version compatibility. Upgrading `client-go` is not needed for this issue, so unless there is another reason I am not aware of, I think this is fine as it is.

In addition, `v1.20.0-rancher.1` is the most recent 1.20.x in the rancher repo.

**Review by cbron — APPROVED** [review]
lgtm, but wait for daishan too

**Review by StrongMonkey — APPROVED** [review]

**Pull request merged**